components/PlayerNameController.cpp

```diff
@@ -0,0 +1,138 @@
+#include "PlayerNameController.h"
+#include "kitten\InputManager.h"
+#include "kitten\K_Instance.h"
+#include "settings_menu\PlayerPrefs.h"
+
+PlayerNameController::PlayerNameController(int p_nameMinLimit, int p_nameMaxLimit) 
+	: 
+	m_nameMinLimit(p_nameMinLimit),
+	m_nameMaxLimit(p_nameMaxLimit),
+	m_update(false)
+{
+
+}
+
+PlayerNameController::~PlayerNameController()
+{
+	input::InputManager::getInstance()->setPollMode(true);
+}
+
+void PlayerNameController::start()
+{
+	m_input = input::InputManager::getInstance();
+	const auto children = getTransform().getChildren();
+
+	kitten::K_GameObject* controllerInput = &children[0]->getAttachedGameObject();
+	m_inputTextbox = controllerInput->getComponent<puppy::TextBox>();
+	m_inputTextbox->setText("");
+	m_stringInputDisplay = controllerInput->getComponent<StringInputDisplay>();
+	m_stringInputDisplay->setCharLimit(m_nameMaxLimit);
+
+	kitten::K_GameObject* confirmButtonGO = &children[1]->getAttachedGameObject();
+	m_confirmButton = confirmButtonGO->getComponent<userinterface::ClickableButton>();
+	m_confirmButtonFrame = confirmButtonGO->getComponent<kitten::ClickableFrame>();
+
+	kitten::K_GameObject* nameStatus = &children[2]->getAttachedGameObject();
+	m_nameStatusTextBox = nameStatus->getComponent<puppy::TextBox>();
+	m_nameStatusTextBox->setText("");
+
+	std::string savedName = PlayerPrefs::getPlayerName();
+	if (savedName != "")
+	{
+		m_playerName = savedName;
+	}
+
+	setConfirmButtonEnabled(false);
+}
+
+void PlayerNameController::onEnabled()
+{
+	if (m_inputTextbox->getText() != m_playerName)
+	{
+		m_inputTextbox->setText(m_playerName);
+	}
+	m_nameStatusTextBox->setText("");
+	setConfirmButtonEnabled(false);
+}
+
+void PlayerNameController::onDisabled()
+{
+	m_update = false;
+	input::InputManager::getInstance()->setPollMode(true);
+}
+
+void PlayerNameController::update()
+{
+	// Using flag to update since setEnabled(false) will stop rendering text
```

Comment:
You can call `setEnabled(false)` on a component to disable it individually instead of `m_attachedObject->setEnabled()` to disable the whole thing and its children